### PR TITLE
SKR networking test

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

File: testing/e2e/skr/skr-networking-test/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    const customParams = {'networking': {'nodes': '10.242.0.0/22'}};
+    const payload = keb.buildPayload('wrong-nodes', 'id01234876', null, null, customParams);
+    const endpoint = `service_instances/id01234876`;
+    const config = await keb.buildRequest(payload, endpoint, 'put');
+
+    try {
+      await axios.request(config);
+      fail('KEB must return an error');
+    } catch (err) {
+      expect(err.response.status).equal(400);
```
Comment (jaroslaw-pieszka) [inline testing/e2e/skr/skr-networking-test/index.js:45]: Do we assert only the status, we do not need to assert returned message?

**Review by jaroslaw-pieszka — DISMISSED** [review]

File: testing/e2e/skr/skr-networking-test/index.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      expect(err.response.data.description).to.include('overlap');
```
Comment (PK85) [inline testing/e2e/skr/skr-networking-test/index.js:46]: @piotrmiskiewicz please print a message to stdout. Any difference in the error message between scenarios "dummy" string VS valid IP range but overlapping?

**Review by jaroslaw-pieszka — DISMISSED** [review]

File: testing/e2e/skr/skr-networking-test/index.js (hunk 2 above)
Comment (piotrmiskiewicz) [reply testing/e2e/skr/skr-networking-test/index.js:46]: done

File: docs/contributor/05-20-e2e_tests.md [hunk 3]
```diff
@@ -100,7 +100,34 @@ export $(xargs < .env)
 make keb-endpoints
 ```
 
-### CI pipelines
+## Networking parameter tests
+
+### Usage
+
+The test executes the following steps:
+1. Calls KEB endpoints with invalid networking parameters
```
Comment (IwonaLanger) [inline docs/contributor/05-20-e2e_tests.md:108]:
```suggestion
1. Calls KEB endpoints with invalid networking parameters.
```

File: docs/contributor/05-20-e2e_tests.md [hunk 4]
```diff
@@ -100,7 +100,34 @@ export $(xargs < .env)
...
+2. Checks whether the call was rejected.
+3. Provisions a cluster with custom networking parameters.
+4. Deprovisions the cluster.
+
+### Test execution
+
+1. Before you run the test, prepare the `.env` file based on the following [`.env.template`](/testing/e2e/skr/skr-networking-test/.env.template):
```
Comment (IwonaLanger) [inline docs/contributor/05-20-e2e_tests.md:115]:
```suggestion
1. Before you run the test, prepare the `.env` file based on this [`.env.template`](/testing/e2e/skr/skr-networking-test/.env.template).
```

File: docs/contributor/05-20-e2e_tests.md [hunk 5]
```diff
@@ -100,7 +100,34 @@ export $(xargs < .env)
...
+
+3. Run the test scenario:
+```bash
+make skr-networking-test
+```
+
+
+
+
+## CI pipelines
```
Comment (IwonaLanger) [inline docs/contributor/05-20-e2e_tests.md]:
```suggestion

## CI pipelines
```

File: testing/e2e/skr/skr-networking-test/index.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+  const customParams = {
+    'networking': {'nodes': '10.253.0.0/21'},
+  };
+  let options = gatherOptions(
+      withCustomParams(customParams),
+  );
+  console.log('Using custom parameters: %o', customParams);
+
+  it('Try networking params which overlaps with restricted IP range', async function() {
```
Comment (IwonaLanger) [inline testing/e2e/skr/skr-networking-test/index.js:35]:
```suggestion
  it('Try networking params which overlap with restricted IP range', async function() {
```

File: testing/e2e/skr/skr-networking-test/index.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+      expect(err.response.status).equal(400);
+    }
+  });
+  it('Perform provisioning', async function() {
+    this.timeout(provisioningTimeout);
+    skr = await getOrProvisionSKR(options, skipProvisioning, provisioningTimeout);
+    options = skr.options;
+  });
+
+  after('Cleanup the resources', async function() {
```
Comment (IwonaLanger) [inline testing/e2e/skr/skr-networking-test/index.js:72]:
```suggestion
  after('Clean up the resources', async function() {
```

**Review by szwedm — DISMISSED** [review]
/lgtm

**Review by IwonaLanger — APPROVED** [review]

**Comment by ralikio** [comment]
Approve of https://github.com/kyma-project/kyma-environment-broker/commit/f626e2f342d298979a64ecf428125f360c911023 a nothing else because of previous @szwedm review.

**Review by ralikio — APPROVED** [review]

**Pull request merged**